metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/parser/AnalysisFileTypeFromSetTest.java

```diff
@@ -0,0 +1,95 @@
+                "        <ASSEMBLY>\n" +
+                "          <STANDARD refname=\"GRCh37\"/>\n" +
+                "        </ASSEMBLY>\n" +
+                "      </SEQUENCE_VARIATION>\n" +
+                "    </ANALYSIS_TYPE>\n" +
+                "    <FILES>\n" +
+                "      <FILE filename=\"UK10K_SCOOP5013826.vcf.gz\" filetype=\"vcf\" checksum_method=\"MD5\" checksum=\"980aad09354c5bc984e23d2f74efdf3b\" unencrypted_checksum=\"641127f130e67cf39a65a5e245c9ecb\"/>\n" +
+                "      <FILE filename=\"ERZ000/ERZ000001/do131_Input_liver_none_mmuC57BL65_CRI01.sra.sorted.bam\" filetype=\"bam\" checksum_method=\"MD5\" checksum=\"15191d68bdd5c1ad23c943c3da3730c7\"/>" +
+                "    </FILES>\n" +
+                "  </ANALYSIS>\n" +
+                "</ANALYSIS_SET>\n";
+
+        AnalysisFileTypeFromSet analysisFileTypeFromSet = new AnalysisFileTypeFromSet();
+        AnalysisSetType analysisSetType = analysisFileTypeFromSet.getAnalysisSet(xmlStr);
+        List<AnalysisFileType> analysisFileList = analysisFileTypeFromSet.extract(analysisSetType);
+        assertAnalysisFileList(analysisFileList);
+    }
+
+    private void assertAnalysisFileList(List<AnalysisFileType> analysisFileList) {
+        assertEquals(analysisFileList.size(), 2);
+        assertEquals(analysisFileList.get(0).getFilename(), "UK10K_SCOOP5013826.vcf.gz");
+        assertEquals(analysisFileList.get(0).getFiletype(), AnalysisFileType.Filetype.Enum.forInt(AnalysisFileType.Filetype.INT_VCF));
+        assertEquals(analysisFileList.get(0).getChecksum(), "980aad09354c5bc984e23d2f74efdf3b");
+        assertEquals(analysisFileList.get(0).getChecksumMethod(), AnalysisFileType.ChecksumMethod.Enum.forString("MD5"));
+        assertEquals(analysisFileList.get(1).getFilename(), "ERZ000/ERZ000001/do131_Input_liver_none_mmuC57BL65_CRI01.sra.sorted.bam");
+        assertEquals(analysisFileList.get(1).getFiletype(), AnalysisFileType.Filetype.Enum.forInt(AnalysisFileType.Filetype.INT_BAM));
+        assertEquals(analysisFileList.get(1).getChecksum(), "15191d68bdd5c1ad23c943c3da3730c7");
+        assertEquals(analysisFileList.get(1).getChecksumMethod(), AnalysisFileType.ChecksumMethod.Enum.forString("MD5"));
+    }
+
+    @Test
+    public void testAnalysisFileParserWrongInput() throws Exception {
+        String xmlStr = "/<?wrong xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+                "<ANALYSIS_SET>\n" +
+                "  <ANALYSIS alias=\"uk10k_scoop5013826.vcf.gz-vcf_analysis-sc-20120330\" center_name=\"SC\" broker_name=\"EGA\" analysis_center=\"SC\" accession=\"ERZ000011\">\n" +
+                "  </ANALYSIS>\n" +
+                "</ANALYSIS_SET>\n";
+
+        AnalysisFileTypeFromSet analysisFileTypeFromSet = new AnalysisFileTypeFromSet();
+        Throwable exception = assertThrows(XmlException.class, () -> analysisFileTypeFromSet.getAnalysisSet(xmlStr));
+        assertEquals("error: Unexpected element: CDATA", exception.getMessage());
```

Comment:
i guess we need to throw/log exceptions with analysis id to refer to the cause of failure

Reply:
done